lib/components/ReceiveFunds.js

```diff
@@ -1,19 +1,59 @@
 import React, { PureComponent } from 'react';
 import { Text, Button, QRCode } from '@bpanel/bpanel-ui';
+import PropTypes from 'prop-types';
+import BoxGrid from './BoxGrid';
+import Label from './Label';
 
 class ReceiveFunds extends PureComponent {
   constructor(props) {
     super(props);
   }
 
+  static get propTypes() {
+    return {
+      receiveAddress: PropTypes.string,
+      createAddress: PropTypes.func,
+      selectedAccount: PropTypes.string,
+      selectedWallet: PropTypes.string,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      receiveAddress: '',
+      selectedAccount: '[select account]',
+      selectedWallet: '[select wallet]',
+    };
+  }
+
+  // generate new receive address
+  async new() {
+    const { createAddress, selectedWallet, selectedAccount } = this.props;
+    await createAddress(selectedWallet, selectedAccount, {
+      addressType: 'receive',
+    });
+  }
+
   render() {
+    const { receiveAddress, selectedAccount, selectedWallet } = this.props;
+    // TODO: allow copy/paste of the address
```

Comment:
adding the `copyIcon` prop I think should allow this out of the box. Might need to do `condensed` too, but use the full length of the string.